### PR TITLE
Multi-word caches + fix parentheses

### PR DESCRIPTION
**Pull request by mikhailshilkov**
https://github.com/pulumi/inflector/pull/3 wasn't enough for multi_word_identifiers, so I'm adding a regex to solve for foo_caches and FooCaches. Also, fixing the `parentheses`.

Now I wonder how many hidden problems we have with multi-word identifiers. For instance, `old_cookies` singularizes to `old_cooky`, while `cookies` translate to `cookie` fine. But that's for another day.

Intend to publish this as v0.2.1 as a bug fix release.

### CODE REVIEW

**Review by iwahbe — APPROVED** [review]
I think this is good.

The more changes we make here, the more robust our backwards compatibility story will need to be to ever upgrade the `inflector` version used by the bridge.

File: inflector.go [hunk 1]
```diff
@@ -196,7 +197,7 @@ func init() {
 			{`(?i)(hive)s$`, `$1`},
 			{`(?i)(drive)s$`, `$1`},
 			{`(?i)(^analy)ses$`, `${1}sis`},
-			{`(?i)(analy|diagno|^ba|(p)arenthe|(p)rogno|(s)ynop|(t)he)ses$`, `${1}${2}sis`},
+			{`(?i)(analy|diagno|^ba|parenthe|progno|synop|the)ses$`, `${1}sis`},
```
Comment (VenelinMartinov) [inline inflector.go:200]: This looks like an unsuccessful attempt to handle capital words - do we handle these well?

**Review by VenelinMartinov — APPROVED** [review]

File: inflector.go (hunk 1 above)
Comment (mikhailshilkov) [reply inflector.go:200]: The `(?i)` modifier makes the whole regex case-insensitive. This test succeeds:

```
{"Parentheses", "Parenthesis", true},
```

Comment (VenelinMartinov) [reply inflector.go:200]: I see so the `(p)` and `(s)` groups were pointless

Comment (mikhailshilkov) [reply inflector.go:200]: Yeah I couldn't guess the motivation. Asked an LLM too but it said it made no sense.

**Pull request merged**